### PR TITLE
Fixing  some of the edge cases with $

### PR DESCRIPTION
**Pull request by dbejar-s**
Some of the edge cases with $ are fixed: echo $, echo"$", echo '$' and so on...

### CODE REVIEW

File: src/clean.c [hunk 1]
```diff
@@ -73,8 +75,25 @@ char	*build_expanded_instruction(char *clean, char *instruction,
 	j = 0;
 	while (instruction[i])
 	{
-		if (instruction[i] == '$' && envir_must_be_expanded(instruction, i))
+		if (instruction[i] == '$' && (instruction[i + 1] == ' ' || instruction[i + 1] == '\0')) //ESTO DEBERIA SER CON  TABULADOR TAMBIEN???
```
Comment (pmarkaide) [inline src/clean.c:78]: You can use ft_isdelim to check for tabs and spaces together

File: src/clean.c [hunk 2]
```diff
@@ -73,8 +75,25 @@ char	*build_expanded_instruction(char *clean, char *instruction,
...
+		{
```
Comment (pmarkaide) [inline src/clean.c:79]: I think handling the edge case inside the function is not the proper way. We can have multiple edge cases that don't fit here. We must create a generic function that expands variables, like is now, and then another check that is run before the expansion to evaluate how to handle that case. As a first thought, we can evaluate using the token value and do normal expansion or edge case expansion

File: src/clean.c [hunk 3]
```diff
@@ -94,10 +113,27 @@ size_t	expanded_envir_len(char *instruction, t_macro *macro)
...
+	// while (instruction[i]) //BORRAR ESTE WHILE, SOLO PRUEBAS
+	// {
+	// 	printf("verdadero o falso: i es%d y char es %c TRUE es 1 y FALSO es O ==>%d\n", i, instruction[i], envir_must_be_expanded(instruction, i));
+	// 	i++;
+	// }
 	i = 0;
 	while (instruction[i])
 	{
-		if (instruction[i] == '$' && envir_must_be_expanded(instruction, i))
+		if (instruction[i] == '$' && (instruction[i + 1] == ' ' || instruction[i + 1] == '\0')) //ESTO DEBERIA SER CON  TABULADOR TAMBIEN???
```
Comment (pmarkaide) [inline src/clean.c:126]: same as before

File: src/clean.c [hunk 4]
```diff
@@ -168,7 +209,7 @@ static int	create_and_add_node(char *start, size_t length, t_list **tokens)
 	return (0);
 }
 
-static int	process_token(char **current_pos, t_list **tokens)
+static int	process_token(char **current_pos, t_list **tokens) //PROBLEMAS AL SEPARAR COMILLAS EN echo hola'mundo' WIP
```
Comment (pmarkaide) [inline src/clean.c:212]: this is correct. `hola'mundo'` is a single argument, as there are no spaces in between the quotes. I can showcase examples later in the meeting

**Review by pmarkaide — CHANGES_REQUESTED** [review]
I don't agree that this approach is the best to handle $? case. We must create a generic envir expansion and handle edge cases outside of it.

**Comment by dbejar-s** [comment]
We will manage the $ cases with a different approach

**Pull request closed without merging**